torchvision/prototype/datasets/__init__.py

```diff
@@ -1,3 +1,13 @@
+try:
+    import torchdata
+except (ModuleNotFoundError, TypeError) as error:
```

Comment:
The `torchdata` library hosted on PyPI fails for me at import with an `TypeError`. This is why we can aggregate both failure cases here.

Reply:
Chatted with @pmeier , this is a bug in the `torchdata` from PyPI (which is unrelated to the `torchdata` we are depending on.

Reply:
If you do `pip install torchdata`, `python -c "import torchdata"` fails with 

```
TypeError: metaclass conflict: the metaclass of a derived class must be a (non-strict) subclass of the metaclasses of all its bases
```